docs/psf_spec/high_level_overview.rst

```diff
@@ -0,0 +1,244 @@
+PSF Fitting
+===========
+
+.. role::  raw-html(raw)
+    :format: html
+
+`photutils` offers a module implementing a series of PSF fitting algorithms. These
+fitting algorithms are broken down into a series of submodules, allowing the user
+to implement specific versions of each part of the fitting process, implementing
+a variety of fitting procedures best suited to the specific science cases. However,
+at its most basic, all choices share the same basic functionality: a 2D image is
+supplied and a `~astropy.table.Table` of detected sources, with positions and flux
+counts, is returned.
+
+This simplified workflow is expanded below for the case of the
+`~photutils.psf.IterativelySubtractedPSFPhotometry` class, building upon the simpler
+`~photutils.psf.BasicPSFPhotometry` algorithm. While there are additional blocks
+used within the workflow, the main `~photutils.psf.BasicPSFPhotometry` flow is
+``image`` :raw-html:`&rarr;` ``finder`` :raw-html:`&rarr;` ``group maker``
+:raw-html:`&rarr;` ``fitter`` :raw-html:`&rarr;` ``stars``. The extension to more
+complex fitting classes, such as `~photutils.psf.IterativelySubtractedPSFPhotometry`,
+have the flow ``image`` :raw-html:`&rarr;` ``finder`` :raw-html:`&rarr;` ``group maker``
+:raw-html:`&rarr;` ``fitter`` :raw-html:`&rarr;` ``culler and ender`` :raw-html:`&rarr;`
+``finder`` :raw-html:`&rarr;` ... :raw-html:`&rarr;` ``stars``.
+
+Each submodule is described briefly here, with context for how it fits into the overall
+PSF fitting framework, providing a top-down view of the process. Where available, links
+to more detailed descriptions of each block is also provided.
+
+.. graphviz::
+
+    digraph PSFFitting {
+        compound = true;
+        newrank = true;
+        subgraph cluster0 {
+            label="IterativelySubtractedPSFPhotometry"
+            f [label = "finder"];
+            g [label = "group maker"];
+            som [label = "single object model"];
+            n [label = "noise"];
+            ft [label = "fitter"];
+            b [label = "background estimator"];
+            c [label = "culler and ender"];
+            // required to allow finder and guess to combine into group maker
+            invis [shape = circle, width = .1, fixedsize = true, style = invis];
+        }
+
+        w [label = "wcs"];
+        i [label = "input image", shape = box];
+        gs [label = "guess at stars", shape = box];
+        p [label = "psf model"];
+        sm [label = "scene maker"];
+        st [label = "stars\n(astropy.table)", shape = box];
```

Comment:
I wonder of "stars" here should be "objects" or something like that?  Presumably the scene maker at least *might* have the option to do other kinds of fitting?

Reply:
I second this also